pkg/runtime/setup.go

```diff
@@ -553,6 +544,33 @@ func (s *setup) uninstall(id strfmt.UUID) (rerr error) {
 		return errs.Wrap(err, "Could not unload artifact envdef")
 	}
 
+	// If this is a dynamically imported artifact, tell the ecosystem to remove/undeploy it.
+	if artifact, exists := s.depot.config.Cache[id]; exists && artifact.Namespace != "" {
```

Comment:
Could you update the depot.Exists() function to return the artifact? I'd prefer we don't access private properties on the depot to keep the responsibilities clear.